rust/src/lib.rs

```diff
@@ -36,10 +42,19 @@ pub extern fn solve_munkres(n: libc::size_t, array: *mut libc::c_double) -> Arra
             res.push(row as libc::c_int);
             res.push(col as libc::c_int);
         }
-        Array::from_vec(res)
+        Box::new(Array::from_vec(res))
     });
     match res {
-        Ok(array) => array,
-        Err(_) => Array::from_vec(vec![-1]),
+        Ok(array) => {
+            Box::into_raw(array)
+        },
+        Err(_) => 0 as *mut Array,
```

Comment:
Minor: `std::ptr::null()`?

Reply:
Still need to cast it to `*mut`:
`std::ptr::null::<Array>() as *mut _`
Not sure if it's much better.

Reply:
Sorry, I should have suggested `std::mem::null_mut()` that avoids any casting and is more explicit.

https://play.rust-lang.org/?gist=1a008ec3d6d7c9b1772118f36e8af59e&version=stable